py/farm_ng/core/events_file_writer.py

```diff
@@ -18,15 +18,25 @@
 
 
 class EventsFileWriter:
-    def __init__(self, file_name: Union[str, Path]) -> None:
+    def __init__(self, file_name: Union[str, Path], max_file_mb: int = 0) -> None:
+        """
+        file_name: Path to and base of file name where the events file will be logged.
+        max_file_mb: Maximum log size in MB. Logging will roll over to new file when reached. Ignored if <= 0.
+        """
         if isinstance(file_name, str):
             file_name = Path(file_name)
-        self._file_name: Path = file_name.absolute()
-        assert Path(self._file_name.parents[0]).is_dir()
+
+        assert Path(file_name.parents[0]).is_dir()
+
+        self._file_base: Path = (file_name.parents[0] / file_name.stem).absolute()
+        self.extension: str = file_name.suffix
```

Comment:
I'd include this as an optional argument and default to ".bin"

In that way it will be more clear that users provide just the session base name and we then create sub files.

Reply:
That's how I had it and changed it in https://github.com/farm-ng/farm-ng-core/pull/58/commits/4a0424d37d2ec1fd6163e7e9029e5c624639fa16. Will revert, as I agree.